include/bluetooth/mesh/access.h

```diff
@@ -641,6 +641,28 @@ int bt_mesh_model_data_store(struct bt_mesh_model *mod, bool vnd,
 int bt_mesh_model_extend(struct bt_mesh_model *mod,
 			 struct bt_mesh_model *base_mod);
 
+/** @brief Register a callback function hook for mesh model messages.
+ *
+ * Register a callback function to act as a hook for recieving mesh model layer messages
+ * directly to the application without having instantiated the relevant models.
+ *
+ * @param cb A pointer to the callback function.
+ */
+void bt_mesh_msg_cb_set(void (*cb)(uint32_t opcode, struct bt_mesh_msg_ctx *ctx,
+			 struct net_buf_simple *buf));
```

Comment:
IMO, we should make this a compile time thing if your use case permits it. 
E.g. make a kconfig option, such as `config BT_MESH_RECV_CB`, and replace the `if (msg_cb)` lines with `if (IS_ENABLED(CONFIG_BT_MESH_RECV_CB))`, then declare the message callback in a header somewhere, so we can see what it should look like.

Reply:
Yes, I like the idea of a compile time check for this. Will do.